cpp/dolfinx/la/PETScMatrix.cpp

```diff
@@ -212,20 +212,20 @@ PETScMatrix::add_fn(Mat A)
 //-----------------------------------------------------------------------------
 std::function<int(std::int32_t, const std::int32_t*, std::int32_t,
                   const std::int32_t*, const PetscScalar*)>
-PETScMatrix::add_block_fn(Mat A)
+PETScMatrix::set_block_fn(Mat A, InsertMode mode)
 {
-  return [A, cache = std::vector<PetscInt>()](
+  return [A, mode, cache = std::vector<PetscInt>()](
              std::int32_t m, const std::int32_t* rows, std::int32_t n,
              const std::int32_t* cols, const PetscScalar* vals) mutable {
     PetscErrorCode ierr;
 #ifdef PETSC_USE_64BIT_INDICES
     cache.resize(m + n);
     std::copy_n(rows, m, cache.begin());
-    std::copy_n(cols, n, cache.begin() + m);
+    std::copy_n(cols, n, std::next(cache.begin(), m));
```

Comment:
@garth-wells Do you know if the `std::next` is needed here?

Reply:
It's seldom really "needed", just a way of showing you are adding to an iterator.